widgetastic_manageiq/__init__.py

```diff
@@ -33,6 +33,57 @@
 from cfme.exceptions import ItemNotFound
 
 
+class SnapshotMemorySwitch(Widget):
+    """ This replaces the old-style checkbox when working with snapshot memory.
+
+    At first glance on the page it looks like a BootstrapSwitch but it really isn't.
+    Basic switch controls are implemented here along with fill and read methods,
+    so this should be usable as any other widget.
+    """
+    LOCATOR = u'//div[contains(@class, "bootstrap-switch-snap_memory")]'
+    ON_LOCATOR = u'//div[contains(@class, "bootstrap-switch-on")]'
+
+    def __init__(self, parent, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+
+    def __locator__(self):
+        return self.LOCATOR
+
+    def switch(self):
+        self.browser.click(self.LOCATOR)
+
+    @property
+    def is_set(self):
+        if self.browser.selenium.find_elements_by_xpath(self.ON_LOCATOR):
+            return True
+        return False
+
+    def switch_on(self):
+        try:
+            if not self.is_set:
+                self.switch()
+        except Exception as e:
```

Comment:
Does this `try` block actually do anything?

Reply:
Sure. If the switch is set (meaning it's ON), it leaves it as is. If the switch is not set (meaning it's OFF), it sets the switch to ON.
So after running the switch_on() function, the switch will always be ON.

Reply:
@apagac I don't think that is quite what @jan-zmeskal  was getting at with his comment.

You have a code block of:
```
try:
    <some_statement>
except:
    raise
```

This is equivalent to:
```
<some_statement>
```

There is nothing done with the exception is caught here, and this try/except block is extraneous.  Is there an exception you were encountering during testing or expecting from the check of is_set or call to self.switch?  

If so those exceptions should be specifically listed here, and then there should at the very least be logging, if not modification of the exception or changes in the functions behavior due to the exception being encountered.

Reply:
There indeed was an exception I was hitting back in the day I wrote this. I'll check for it and fix.